src/SpaceCat/features.py

```diff
@@ -0,0 +1,470 @@
+        grouped_cell_freq_region.columns = ['value']
+        grouped_cell_freq_region.reset_index(inplace=True)
+        grouped_cell_freq_region['metric'] = 'total_cell_freq'
+        grouped_cell_freq_region.rename(columns={self.compartment_col: 'subset'}, inplace=True)
+        grouped_cell_freq_region['cell_type'] = 'all'
+
+        # add manually defined dfs to overall list
+        cluster_dfs.extend([grouped_cell_counts,
+                            grouped_cell_counts_region,
+                            grouped_cell_freq_region])
+
+        # create single df with appropriate metadata
+        total_df_clusters = pd.concat(cluster_dfs, axis=0)
+
+        # compute density of cells for counts-based metrics
+        count_metrics = total_df_clusters.metric.unique()
+        count_metrics = [x for x in count_metrics if 'count' in x]
+
+        count_df = total_df_clusters.loc[total_df_clusters.metric.isin(count_metrics), :]
+        compartment_area_df = compartment_area_df.rename(columns={self.compartment_col: 'subset'})
+        all_area = compartment_area_df[[self.image_col, self.compartment_col + '_area']].groupby(
+            by=[self.image_col]).sum().reset_index()
+        all_area['subset'] = 'all'
+        area_df = pd.concat([compartment_area_df, all_area])
+        count_df = count_df.merge(area_df, on=[self.image_col, 'subset'], how='left')
+        count_df['value'] = count_df['value'] / count_df[self.compartment_col + '_area']
+        count_df['value'] = count_df['value'] * 1000
+
+        # rename metric from count to density
+        count_df['metric'] = count_df['metric'].str.replace('count', 'density')
+        count_df = count_df.drop(columns=[self.compartment_col + '_area'])
+        total_df_clusters = pd.concat([total_df_clusters, count_df], axis=0)
+
+        if exclude_missing_compartments:
+            total_df_clusters = self.exclude_empty_compartments(total_df_clusters)
+
+        self.adata_table.uns['cluster_stats'] = total_df_clusters
+
+    def format_helper(self, compartment_df, compartment, cell_pop_level, feature_type):
+        """ Add informative metadata columns to the feature dataframe.
+        Args:
+            compartment_df (pd.DataFrame): table with features for the specified compartment
+            compartment (str): which compartment is being subsetted
+            cell_pop_level (str): level of cell clustering for the feature
+            feature_type (str): broad name of the feature type
+
+        Returns:
+            pd.DataFrame:
+                table with feature metadata columns included
+        """
+        # add useful metadata about the feature
+        if compartment == 'all':
+            compartment_df['feature_name_unique'] = compartment_df['feature_name']
+        else:
+            compartment_df['feature_name_unique'] = compartment_df['feature_name'] + '__' + \
+                                                    compartment
```

Comment:
If you really wanted to:

```suggestion
        compartment_df["feature_name_unique"] = compartment_df["feature_name"]
        if compartment != "all":
            compartment_df["feature_name_unique"] = compartment_df["feature_name_unique"] + "__" + compartment
```